lightly/models/modules/masked_vision_transformer_timm.py

```diff
@@ -0,0 +1,140 @@
+from typing import Optional, Union
+
+import torch
+from timm.layers.pos_embed import resample_abs_pos_embed
+from timm.models.vision_transformer import VisionTransformer
+from torch import Tensor
+from torch.nn import Parameter
+
+from lightly.models import utils
+from lightly.models.modules.masked_vision_transformer import MaskedVisionTransformer
+
+import torch
+import torch.nn as nn
+from torch.nn import LayerNorm, Linear, Module, Parameter, Sequential
+
+
+class MaskedVisionTransformerTIMM(MaskedVisionTransformer):
+    def __init__(
+        self,
+        vit: VisionTransformer,  # here assume that we always initialize with a vit model from timm
+        mask_token: Union[bool, Parameter],
+        device : str
+    ):
+        super().__init__()
+        self.vit = vit
+        self.mask_token = (
+            mask_token
+            if isinstance(mask_token, Parameter)
+            else Parameter(torch.zeros(1, 1, vit.num_features))
+            if mask_token
+            else None
+        )
+        self.sequence_length = vit.patch_embed.num_patches + vit.num_prefix_tokens
+        self.device = device
+
+    def forward(
+        self,
+        images: Tensor,
+        idx_mask: Optional[Tensor] = None,
+        idx_keep: Optional[Tensor] = None,
+    ) -> Tensor:
+        out = self.encode(images, idx_keep, idx_mask)
+        class_token = out[:, 0]
+        return class_token
```

Comment:
```suggestion
         if self.vit.attn_pool is not None:
            x = self.vit.attn_pool(x)
        elif self.vit.global_pool == 'avg':
            x = x[:, self.vit.num_prefix_tokens:].mean(dim=1)
        elif self.vit.global_pool:
            x = x[:, 0]  # class token
```
We should handle the different cases as in VisionTransformer:
https://github.com/huggingface/pytorch-image-models/blob/53a48883280743f5c878897bdfd3d471f1c191e4/timm/models/vision_transformer.py#L681-L686